cmd/kubeadm/app/constants/constants.go

```diff
@@ -32,48 +32,82 @@ import (
+	ServiceAccountPublicKeyName = "sa.pub"
+	// ServiceAccountPrivateKeyName defines SA private key base name
 	ServiceAccountPrivateKeyName = "sa.key"
 
+	// FrontProxyCACertAndKeyBaseName defines front proxy CA certificate and key base name
 	FrontProxyCACertAndKeyBaseName = "front-proxy-ca"
-	FrontProxyCACertName           = "front-proxy-ca.crt"
-	FrontProxyCAKeyName            = "front-proxy-ca.key"
+	// FrontProxyCACertName defines front proxy CA certificate name
+	FrontProxyCACertName = "front-proxy-ca.crt"
+	// FrontProxyCAKeyName defaines front proxy CA key name
+	FrontProxyCAKeyName = "front-proxy-ca.key"
 
+	// FrontProxyClientCertAndKeyBaseName defines front proxy certificate and key base name
 	FrontProxyClientCertAndKeyBaseName = "front-proxy-client"
-	FrontProxyClientCertName           = "front-proxy-client.crt"
-	FrontProxyClientKeyName            = "front-proxy-client.key"
-	FrontProxyClientCertCommonName     = "front-proxy-client" //used as subject.commonname attribute (CN)
-
-	AdminKubeConfigFileName             = "admin.conf"
-	KubeletBootstrapKubeConfigFileName  = "bootstrap-kubelet.conf"
-	KubeletKubeConfigFileName           = "kubelet.conf"
+	// FrontProxyClientCertName defines front proxy certificate name
+	FrontProxyClientCertName = "front-proxy-client.crt"
+	// FrontProxyClientKeyName defines front proxy key name
+	FrontProxyClientKeyName = "front-proxy-client.key"
+	// FrontProxyClientCertCommonName defines front proxy certificate common name
+	FrontProxyClientCertCommonName = "front-proxy-client" //used as subject.commonname attribute (CN)
+
+	// AdminKubeConfigFileName defines kuberenetes config file name
+	AdminKubeConfigFileName = "admin.conf"
+	// KubeletBootstrapKubeConfigFileName defines kubernetes bootstrap config file
+	KubeletBootstrapKubeConfigFileName = "bootstrap-kubelet.conf"
+
+	// KubeletKubeConfigFileName defines kubelet's confg file name
+	KubeletKubeConfigFileName = "kubelet.conf"
+	// ControllerManagerKubeConfigFileName defines controller manager's config file name
 	ControllerManagerKubeConfigFileName = "controller-manager.conf"
-	SchedulerKubeConfigFileName         = "scheduler.conf"
+	// SchedulerKubeConfigFileName defines scheduler's config file name
+	SchedulerKubeConfigFileName = "scheduler.conf"
 
 	// Some well-known users and groups in the core Kubernetes authorization system
 
-	ControllerManagerUser   = "system:kube-controller-manager"
-	SchedulerUser           = "system:kube-scheduler"
-	MastersGroup            = "system:masters"
-	NodesGroup              = "system:nodes"
+	// ControllerManagerUser defines well-known user
+	ControllerManagerUser = "system:kube-controller-manager"
+	// SchedulerUser defines well-known user
+	SchedulerUser = "system:kube-scheduler"
+	// MastersGroup defines well-known group
+	MastersGroup = "system:masters"
+	// NodesGroup defines well-known group
+	NodesGroup = "system:nodes"
+	// NodesClusterRoleBinding defines well-known group
```

Comment:
defines the well-known ClusterRoleBinding which binds the too permissive `system:node` ClusterRole to the `system:nodes` group. Since kubeadm is using the Node Authorizer, this ClusterRoleBinding's `system:nodes` group subject is removed if present.